### ISSUE TITLE
Needs TypeScript definition file

### ISSUE THREAD

**Issue by stuft2** [byu-oit/openapi-enforcer#55]

**Comment by Gi60s** [comment]
I've begun work on this.

**Comment by malkevych** [comment]
Any progress?

**Comment by Gi60s** [comment]
Yes. I've been rewriting the library in typescript. It's not the fastest way to get to a TypeScript definition file though so I might have to rethink the approach.

**Comment by sempasha** [comment]
@Gi60s do you need some help on this?
Whould you share your work at some git branch?

**Comment by Gi60s** [comment]
@sempasha I'm always willing to accept some help and feedback. As far as receiving those changes I'm more accustomed to forks and pull requests. I didn't even know it was possible to share a single branch.

Would you be willing to try the fork and PR method?

**Comment by sempasha** [comment]
> Or, are you referring to helping on version 2 with Typescript

@Gi60s, that is exactly what I mean. You said you rewriting library in typescript. And I'm asking you to share your work on this at some branch, to let community help you with this.

**Comment by Gi60s** [comment]
@sempasha I will share it soon. It's far from complete but I'm cleaning up a few loose ends, then I'll share so other can contribute. Hopefully tomorrow.

**Comment by olof-nord** [comment]
Happy to hear some updates! We have used this library a while now, really appreciate your work.

**Comment by Gi60s** [comment]
Thank you @olof-nord. I appreciate the support that many have given to make this project a success.

For anyone that is interested in helping to shape version 2, please see the [newly opened discussion board](https://github.com/byu-oit/openapi-enforcer/discussions/107).

**Comment by zacharytwhite** [comment]
Any updates on this?

**Comment by Gi60s** [comment]
Yes, there are updates. I'm actively working on v2 on this branch: https://github.com/Gi60s/openapi-enforcer/tree/v2/src

v2 will have these features and a few more I'm sure:

* Written in TypeScript 🎉 
* Will be more flexible with error escalation and de-escalation
* Will include the file path and line numbers in errors
* Will run in the browser and in Node
* Will support multiple languages. Although the initial release will likely just be English, it's built to easily extend to additional languages.

If anyone is interested in helping to write v2 then we should talk.
